docs/source/running-jobs/distributed-jobs.rst

```diff
@@ -62,17 +61,32 @@ SkyPilot exposes these environment variables that can be accessed in a task's ``
   task; the same as the count in ``accelerators: <name>:<count>`` (rounded up if a fraction).
 
 
-Executing a multi-node task
+Launching a multi-node task (new cluster)
 -----------------------------------------
 
-The execution behavior is: all nodes are provisioned (barrier);
-workdir/file_mounts are synced to all nodes (barrier); ``setup`` commands are
-executed on all nodes (barrier); finally, the ``run`` commands are executed on
-all nodes.
+When using ``sky launch`` to launch a multi-node task on **a new cluster**, the following happens in sequence:
 
+1. Nodes are provisioned. (barrier)
+2. Workdir/file_mounts are synced to all nodes. (barrier)
+3. ``setup`` commands are executed on all nodes. (barrier)
+4. ``run`` commands are executed on all nodes.
 
-To execute a command on the head node only (common for tools like ``mpirun``),
-use ``SKY_NODE_RANK`` as follows:
+Launching a multi-node task (existing cluster)
+-----------------------------------------
+
+When using ``sky launch`` to launch a multi-node task on **an existing cluster**, the cluster may have more nodes than the current task's ``num_nodes`` requirement.
+
+The following happens in sequence:
+
+1. SkyPilot checks the runtime on all nodes are up-to-date. (barrier)
+2. Workdir/file_mounts are synced to all nodes. (barrier)
+3. ``setup`` commands are executed on **all nodes** of the cluster. (barrier)
+4. ``run`` commands are executed on **the subset of nodes** scheduled to execute the task, which may be fewer than the cluster size.
+
```

Comment:
It is a bit scary to read that setup is run on **all nodes** ("what if my new setup conflicts with tasks already running?"). Should we add a tip box here reminding users that if you want to skip setup, you can use `sky exec` to execute just the run command?

Reply:
Good catch, done.